Update docs & extend docs for test setup

mmeijerdfki opened the pull request:
Closes #368.

Changes to highlight for discussion:

- removal of code in the docs, instead linking the code snippets using [pymdown-extensions](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) in conjunction with raw.githubusercontent.com.
- Addition of the `.drawio` file of the diagram to explain the message abstractions into the source control.

The first point brings the pro that updates are directly reflected (when the docs are updated through mkdocs), but imposes the problem that snippets are specified using line numbers and those possibly would've to be updated as well.

Mark-Niemeyer commented on docs/mkdocs/reference/pytests-message-abstractions.md:
```diff
@@ -0,0 +1,143 @@
+
+Now the assembled datatype can be accessed using the property `queryinst_builder.datatype_instance` and
+the underlying flatbuffers builder can be accessed by using `queryinst_builder.builder` for further use,
+e.g. like in this case calling the service function `call_get_instances_fb()` using the [ServiceManager](http://127.0.0.1:8001/reference/python-helpers/#service_managerpy).
```
Make this a relative path. This is the path from your local test deployment.

Mark-Niemeyer commented:
> but imposes the problem that snippets are specified using line numbers and those possibly would've to be updated as well.

We could add comments to the python files so that the person who wants to change them is informed that it effects the docs

mmeijerdfki commented on .pre-commit-config.yaml:
```diff
@@ -27,6 +27,7 @@ repos:
       - id: check-xml
       - id: check-yaml
         args: [--allow-multiple-documents]
+        exclude: mkdocs.yml
```
The checker doesn't allow for custom tags as discussed in the issue at https://github.com/pre-commit/pre-commit-hooks/issues/103. 
The same problem as in [this issue](https://github.com/pre-commit/pre-commit-hooks/issues/567) occurs.
For the extensions we use the same custom tags functionality:
https://github.com/agri-gaia/seerep/blob/7dddbf7af2b0336509e8efad95f03ed5e01b802a/mkdocs.yml#L59-L61

Now another workaround is implemented, which uses the `--unsafe` option just for the `mkdocs.yml`.

mmeijerdfki commented:
I added a notice when files are referenced in the `mkdocs`. I also updated some links in the docs, such that `mkdocs` is happier and doesn't print as many messages as before.
